src/Symfony/Component/Finder/Tests/FinderTest.php

```diff
@@ -345,6 +362,33 @@ public function testExclude()
         $this->assertIterator($this->toAbsolute([
             'test.php',
             'test.py',
+            'top',
+            'toto',
+            'foo bar',
+            'qux',
+            'qux/baz_100_1.py',
+            'zebulon.php',
+            'Zephire.php',
+            'qux/baz_1_2.py',
+            'qux_0_1.php',
+            'qux_1000_1.php',
+            'qux_1002_0.php',
+            'qux_10_2.php',
+            'qux_12_0.php',
+            'qux_2_0.php',
+        ]), $finder->in(self::$tmpDir)->getIterator());
+    }
+
+    public function testExcludeTopLevelOnly()
```

Comment:
This is the new test case. The top-level `foo` has been excluded, but `top/foo` and `top/foo/file.tmp` are still included, which is what we want the new feature to do.